bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,61 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn bn_mul(bn: U256, d: u64) -> (b256, u64) {
+    disable_panic_on_overflow();
+    let tuple_result: (b256, u64) = (
+        0x0000000000000000000000000000000000000000000000000000000000000000,
+        0,
+    );
+    let product_buffer = asm(bn: __addr_of(bn), d: d, c0, c1, v, f, product, carry_offset, product_buffer: __addr_of(tuple_result)) {
+        // Run multiplication on the lower 64bit word
+        lw v bn i3; // load the word in (bn + 3 words) into v
+        mul v v d; // multiply v * d and save result in v
+        move c1 of; // record the carry
+        sw product_buffer v i3; // store the word in v in product_buffer + 3 words
+        // Run multiplication on the next 64bit word
+        lw v bn i2;
+        mul v v d;
+        move c0 of;
+        add v v c1; // add the previous carry
+        add c1 c0 of; // record the total new carry
+        sw product_buffer v i2;
+
+        // Run multiplication on the next 64bit word
+        lw v bn i1;
+        mul v v d;
+        move c0 of;
+        add v v c1; // add the previous carry
+        add c1 c0 of; // record the total new carry
+        sw product_buffer v i1;
+
+        // Run multiplication on the next 64bit word
+        lw v bn i0;
+        mul v v d;
+        move c0 of;
+        add v v c1; // add the previous carry
+        add c1 c0 of; // record the total new carry
+        move c0 of;
+        sw product_buffer v i0;
+
+        // add address of product and 4 words/32 bytes, store in carry_offset
+        addi carry_offset product_buffer i32;
+        sw carry_offset c1 i0;
```

Comment:
Can't you just use the immediate in the `sw` opcode instead of using a new `carry_offset` register and and `addi` operation? Also, at this point in the code, you can freely use the `v`, `d`, or `c1` registers since they're no longer needed (instead of using a new `carry_offset` register)